R/make-request.R

```diff
@@ -0,0 +1,39 @@
+#' Make a Google API request
+#'
+#' Intended primarily for internal use in client packages that provide
+#' high-level wrappers for users. `make_request()` does very little: calls an
+#' HTTP method, only adding a user agent. Typically the input is created with
+#' [build_request()] and the output is processed with `process_response()`.
+#'
+#' @param x List. Holding the components for an HTTP request, presumably created
+#'   with [develop_request()] or [build_request()]. Must contain the a `method`
+#'   and `url`. If present, `body` and `token` are used.
+#' @param user_agent A user agent string, prepared by [httr::user_agent()].
+#' @param ... Optional arguments passed through to the HTTP method.
+#'
+#' @return Object of class `response` from [httr].
+#' @export
+#' @family requests and responses
+make_request <- function(x, ..., user_agent = gargle_user_agent()) {
+  method <- list(
```

Comment:
A `switch()` would be slightly more elegant here and would make it easier to throw an error if the method did not match.

Reply:
done